job-service/src/main/java/com/hpe/caf/services/job/api/JobsPut.java

```diff
@@ -162,13 +164,6 @@ public static String createOrUpdateJob(final String partitionId, String jobId, N
                 throw new BadRequestException(ERR_MSG_TASK_DATA_DATATYPE_ERROR);
             }
 
-            //  Load serialization class.
-            Codec codec = ModuleLoader.getService(Codec.class);
-
-            //  Get app config settings.
-            LOG.debug("createOrUpdateJob: Reading database and RabbitMQ connection properties...");
-            AppConfig config = AppConfigProvider.getAppConfigProperties();
```

Comment:
The idea behind it was to help reducing the number of parameters passed between functions. I don't get what the downside would be in case of error? More risk of error if we call the function several times instead of storing into a parameter and pass it along?

Reply:
The problem is the context in which it is initialized.  By doing this you moved initialization from when the `createOrUpdateJob()` call is made to a time before any of the class methods are called (possibly even to service startup).  When it was when the `createOrUpdateJob()` call is made the error could be bubbled back to the client whereas that cannot happen during class initialization.  If you just want to reduce the parameters between functions without changing the initialization context then one way would be to use a nested class.  For example [`RandomSamplingWorker.Task`](https://github.houston.softwaregrp.net/Verity/worker-sampling/blob/v1.1.1/worker-sampling/src/main/java/com/microfocus/apollo/workers/sampling/RandomSamplingWorker.java#L101) takes that approach.  Note that the parameters are grouped into a nested object.  They are then used by [`process()`](https://github.houston.softwaregrp.net/Verity/worker-sampling/blob/v1.1.1/worker-sampling/src/main/java/com/microfocus/apollo/workers/sampling/RandomSamplingWorker.java#L141), [`getSamplingDispatchJobParametersDigest()`](https://github.houston.softwaregrp.net/Verity/worker-sampling/blob/v1.1.1/worker-sampling/src/main/java/com/microfocus/apollo/workers/sampling/RandomSamplingWorker.java#L232), [`createSamplingDispatchJobParameters()`](https://github.houston.softwaregrp.net/Verity/worker-sampling/blob/v1.1.1/worker-sampling/src/main/java/com/microfocus/apollo/workers/sampling/RandomSamplingWorker.java#L248), etc., all without being repeatedly passed around.

Reply:
Oh ok, thanks for the explanation.